wavefront_python_sdk/common/utils.py

```diff
@@ -0,0 +1,260 @@
+"""
+Utils module contains useful function for preparing and processing data.
+
+@author: Hao Song (songhao@vmware.com)
+"""
+
+import re
+import io
+from gzip import GzipFile
+import threading
+
+
+# pylint: disable=too-few-public-methods
+
+class AtomicCounter(object):
+    """An atomic, thread-safe incrementing counter."""
+
+    def __init__(self, initial=0):
+        """
+        Construct Atomic Counter.
+
+        @param initial: Initial value of the counter
+        """
+        self.value = initial
+        self._lock = threading.Lock()
+
+    def increment(self, num=1):
+        """
+        Increment atomic counter value.
+
+        @param num: Num to be increased, 1 by default
+        @return: Current value after increment
+        """
+        with self._lock:
+            self.value += num
+            return self.value
+
+
+def chunks(data_list, batch_size):
+    """
+    Split list of data into chunks with fixed batch size.
+
+    @param data_list: List of data
+    @param batch_size: Batch size of each chunk
+    @return: Return a lazy generator object for iteration
+    """
+    for i in range(0, len(data_list), batch_size):
+        yield data_list[i:i + batch_size]
+
+
+def gzip_compress(data, compresslevel=9):
+    """
+    Compress data using GZIP.
+
+    @param data: Data to compress
+    @param compresslevel: Compress Level
+    @return: Compressed data
+    """
+    buf = io.BytesIO()
+    with GzipFile(fileobj=buf, mode='wb', compresslevel=compresslevel) \
+            as gzip_file:
+        gzip_file.write(data)
+    return buf.getvalue()
+
+
+def sanitize(string):
+    """
+    Sanitize a string, replace whitespace with "-".
+
+    @param string: Input string
+    @return: Sanitized string
+    """
+    whitespace_sanitized = re.sub(r"[\s]+", "-", string)
```

Comment:
Looks like `re.sub` here might be replaced with more efficient `'-'.join(string.split())`

```
$ python3 -m timeit -s "import re; test = 'abc def xyz'" -c 're.sub(r"[\s]+", "-", test)'
1000000 loops, best of 3: 1.77 usec per loop
```
5x faster
```
$ python3 -m timeit -s "test = 'abc def xyz'" -c "'-'.join(test.split())"
1000000 loops, best of 3: 0.364 usec per loop
```

Reply:
It will even pass your current test cases, but will produce a slightly different results in case like
```
>>> string = '\n hello \n \t world \n '
>>> re.sub(r"[\s]+", "-", string)
'-hello-world-'
>>> 
>>> '-'.join(string.split())
'hello-world'
```

Reply:
Please double check what the expected behavior should be and may be add more cases to your unittest